library/ssl_tls.c

```diff
@@ -2852,6 +2893,8 @@ int mbedtls_ssl_encrypt_buf( mbedtls_ssl_context *ssl,
             return( ret );
         }
 
+        encryption_status = ENCRYPTION_SUCCESS;
```

Comment:
@AndrzejKurek , may be better to implement it this way ?
encryption_status = mbedtls_cipher_crypt( &transform->cipher_ctx_enc,
                                   transform->iv_enc,	                                   transform->iv_enc,
                                   transform->ivlen,	                                   transform->ivlen,
                                   data, rec->data_len,	                                   data, rec->data_len,
                                   data, &olen ) ) != 0 )	                                   data, &olen ) ) != 0 );

this decrease the chance an attacker can skip   mbedtls_cipher_crypt() and jump directly to 
 encryption_status = ENCRYPTION_SUCCESS; line. 
I understand that you wanted to prevent encryption_status to be 0 as success value, but it seems that skip still possible this way

Reply:
Added.